modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/adapters/BaseNuxeoArtifactDocAdapter.java

```diff
@@ -203,4 +208,15 @@ protected static DocumentModelList query(CoreSession session, String query) {
         }
     }
 
+    protected static List<String> queryAndFetchIds(CoreSession session, String idProp, String type,
+            DocumentModel parent, String order) {
+        String query = QueryHelper.select(idProp, type, parent, order);
+        PartialList<Map<String, Serializable>> res = session.queryProjection(query, ArtifactSearcher.MAX_RESULTS, 0);
+        return res.stream()
+                  .map(e -> e.get(idProp))
+                  .filter(String.class::isInstance)
+                  .map(String.class::cast)
```

Comment:
Do you expect to have non-String ids? Worth adding a comment why if it's the case. Normal doc ids are always strings in Nuxeo. And I think your other ids are also strings too.

Reply:
Everithing should be strings here indeed: you meam i should cast directly and remove the isInstance check?

Reply:
(but just in case: we're casting the prop id that can be specific to the doc type and is given in parameter: this is not a Nuxeo "repo doc id")

Reply:
removed isInstance to cast directly